### ISSUE TITLE
Find way to check for Circulant graphs

### ISSUE THREAD

**Issue by MuellMark** [MuellMark/DDMGraphLabelingGenerator#35]
This issue is stemming from #34. Basically, before we can go forward in generating these graphs, then need to be checked for first. This issue if for coming up with the way to check them, while the previous issue is a larger thread to plan out the whole issue

**Comment by MuellMark** [comment]
I need to add methods to make it easier to call the different test functions. I have a true false but now I have 3 methods so I'll need to expand this before going forward

**Comment by MuellMark** [comment]
Added a method selector for this and #32 and any other future methods. I will now focus on finding cycles to then check is they are the circulant graphs

**Comment by MuellMark** [comment]
A few things to note: the last 3 commits were accidentally labeled as #4 because I've been working on a different repo on issue #4, and I only caught this mistake now. Secondly, after having made similar methods multiple times, it is clear to me that some new copy methods will need to be implemented before I can make this all work. I can see I'm improving as I can see these issues ahead of time, but that doesn't mean it will be trivial to code in.

**Comment by MuellMark** [comment]
I was foolishly optimistic that my code would work without copy functions, One is added, but I had to try it out regardless. Shifting gears to make sure it works

**Comment by MuellMark** [comment]
I'm going to take a step back and work on something else until I have more time to sit down and focus on the recursive method uninterupted I'm going to work on #33

**Comment by MuellMark** [comment]
I added debugging statements and got rid of the infinite loop error. However, there is another issue because it is finding a cycle that is repeating edges, and the cycle specified must have all edges in it only once. I'm investigating further

**Comment by MuellMark** [comment]
There are still some issue with the recursive method, as it is producing invalid cycles. I'll investigate further, but this shouldn't be more restrictive than making sure all vertices have 4 edges.

Part of me feels like its an issue with how the method is being called, but the method itself should not be creating invalid cycles in the first place

**Comment by MuellMark** [comment]
I got the check for cycles working. This won't actually be used to exclude graphs, as all of these graphs should contain cycles. Rather, it gets all cycles to then be tested if they are circulent

**Comment by MuellMark** [comment]
As seen in my last commit, none of the generated graphs with 5 vertices work for circulant graphs. However, that doesn't mean that there isn't a labeling that does work with circulant graphs. that means part of #32 needs to be done to completely test this, however, I can still move forwards to see if any of the 4 6 vertex graphs works for this

**Comment by MuellMark** [comment]
I added in the last method for the checker, but it said none of the 6 graphs worked. I'll try with 7 graphs, but if that doesn't work I'll create a separate graph that is circulant to check

**Comment by MuellMark** [comment]
Added a Circulant test graph, and it didn't define it as such, investigating further

**Comment by MuellMark** [comment]
I found the issue, I got some of the indices mixed up so different parts were called at inaccurate times. I'll run some more tests and clean up the code before figuring out my next steps

**Comment by MuellMark** [comment]
I'm currently trying to check if the graphs generated that say they are circulant actually are, however, in doing so I found that  printing them as an adjacency matrix no longer works, as the 3's are completely filled out despite not actually having all of those numbers

**Comment by MuellMark** [comment]
I can safely say that I have a way of checking for Cn(1,2) graphs. Some better comments should be made but there are some other things to add, so I may save it for then. As it stands, there are 4 graphs that are both ddm and circulant from the 6 vertex graphs, but I need a better way of visualizing them to showcase them first. I'll keep working to generate more
